### PR TITLE
Vue: fix various model/input/state issues

### LINKED ISSUE #4879: [3.8.3-rc.1] Structure field values do not save
## Description
Structure field items can be added/removed/sorted, but item field values are not saved. Structure fields operate normally in 3.8.2

**Expected behavior**  
When adding/updating a structure field item, item field values should be updated.

## To reproduce
Add a structure field to a page blueprint and create an item. Item should be added, but field value will not be saved. Testing with the following structure field definition:

```yaml
teststructure:
  type: structure
  fields:
    title:
      type: text
```

## Your setup
**Kirby Version**  
3.8.3-rc.1

**Your system (please complete the following information)**
 - Device: MacBook Pro
 - OS: macOS
 - Browser: Chrome
 - Version: 107

### LINKED ISSUE #4881: [3.8.3-rc.1] Avoid mutating `value` prop directly
1. Demokit, go to http://sandbox.test/panel/pages/agency+team+janae-randolph
2. Change email address
3. Error in console:

```
vue.esm.js:5103 [Vue warn]: Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed property based on the prop's value. Prop being mutated: "value"

found in

---> <KForm> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Forms/Form.vue
       <KFieldsSection> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Sections/FieldsSection.vue
         <KColumn> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Layout/Column.vue
           <KGrid> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Layout/Grid.vue
             <KSections> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Sections/Sections.vue
               <KView> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Layout/View.vue
                 <KPanel> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Layout/Panel.vue
                   <KInside> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Layout/Inside.vue
                     <KPageView> at /Users/nhoffmann/Sites/kirby/sandbox/kirby/panel/src/components/Views/PageView.vue
                       <Fiber>
                         <Root>
```

### PR DESCRIPTION
**Pull request by distantnative**
### Fixed regression
- Fixed structure fields not updating their local model state
https://github.com/getkirby/kirby/issues/4879
- Fixed form dialogs not updating their locale values state
- `k-form` doesn't complain anymore about directly mutating the `value` prop
https://github.com/getkirby/kirby/issues/4881

## Ready?

- [x] ~~Unit tests for fixed bug/feature~~
- [x] ~~In-code documentation (wherever needed)~~
- [x] Tests and checks all pass

### For review team

- [x] ~~Add changes to release notes draft in Notion~~
- [x] ~~Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)~~

### CODE REVIEW

**Comment by distantnative** [comment]
@afbora would you mind help testing https://github.com/getkirby/kirby/issues/4881 and https://github.com/getkirby/kirby/issues/4879 against this PR?

**Comment by afbora** [comment]
Great! #4879 issue seems fixed with the PR.

File: panel/src/components/Forms/Field/DateField.vue [hunk 1]
```diff
@@ -95,7 +95,7 @@ import { props as DateInput } from "../Input/DateInput.vue";
  * Have a look at `<k-field>`, `<k-input>`
  * and `<k-datetime-input>` for additional information.
  *
- * @example <k-date-field :value="date" name="date" label="Date" @input="value = $event" />
+ * @example <k-date-field :value="date" name="date" label="Date" @input="this>$emit('input', $event)" />
```
Comment (lukasbestle) [inline panel/src/components/Forms/Field/DateField.vue]:
```suggestion
 * @example <k-date-field :value="date" name="date" label="Date" @input="$emit('input', $event)" />
```

**Review by bastianallgeier — APPROVED** [review]
LGTM. Just checked and #4881 is fixed as well.

**Pull request merged**